manager/orchestrator/globaljob/orchestrator.go

```diff
@@ -54,6 +67,136 @@ func (o *Orchestrator) run(ctx context.Context) {
 	<-o.stopChan
 }
 
+// reconcileService determines whether a service has enough tasks, and creates
+// more if needed.
+func (o *Orchestrator) reconcileService(id string) error {
+	var (
+		service *api.Service
+		tasks   []*api.Task
+		nodes   []*api.Node
+	)
+
+	// we need to first get the latest iteration of the service, its tasks, and
+	// the nodes in the cluster.
+	o.store.View(func(tx store.ReadTx) {
+		service = store.GetService(tx, id)
+
+		// getting tasks with FindTasks should only return an error if we've
+		// made a mistake coding; there's no user-input or even reasonable
+		// system state that can cause it. If it returns an error, we'll just
+		// panic and crash.
+		var err error
+		tasks, err = store.FindTasks(tx, store.ByServiceID(id))
+		if err != nil {
+			panic(fmt.Sprintf("error getting tasks: %v", err))
+		}
+
+		// same as with FindTasks
+		nodes, err = store.FindNodes(tx, store.All)
+		if err != nil {
+			panic(fmt.Sprintf("error getting nodes: %v", err))
+		}
+	})
+
+	if service == nil {
```

Comment:
Nit: could arguably return early from the `o.store.View` func in that case, save a little time?